lib/ElmCompiler.js

```diff
@@ -44,10 +43,6 @@ function runCompiler(sources, options) {
     options.processOpts
   );
 
-  if (options.verbose) {
-    console.log(['Running', pathToElm].concat(processArgs).join(' '));
-  }
-
```

Comment:
Just curious, why remove?

Reply:
This was triggered via the `--verbose` CLI flag, but that flag:

- wasn’t documented
- had no tests
- wasn’t even mention in the minimst options
- was not very useful

Before replacing the minimst stuff, I didn’t even know there was a `--verbose` thing. So I tried it out for the first time. And was I disappointed! I expected the output to be really verbose with lots of details about what’s going on. But no. All it does is logging which `elm make` calls are performed.

That could potentially have been at least somewhat interesting in the past, when we ran the compiler 2 to 3 times per `elm-test` execution (2 times for applications, 3 times for packages). That could look like this:

```
❯ npx elm-test --verbose
Running /Users/lydell/forks/elm-review/node_modules/.bin/elm make --output /dev/null
Running /Users/lydell/forks/elm-review/node_modules/.bin/elm make /Users/lydell/forks/elm-review/tests/Documentation/ReadmeLinksPointToCurrentVersion.elm /Users/lydell/forks/elm-review/tests/Documentation/ReadmeLinksPointToCurrentVersionTest.elm /Users/lydell/forks/elm-review/tests/Fixtures/Dependencies.elm /Users/lydell/forks/elm-review/tests/MiscRules/NoHtmlButton.elm /Users/lydell/forks/elm-review/tests/MiscRules/NoHtmlButtonTest.elm /Users/lydell/forks/elm-review/tests/MiscRules/NoUnusedTypeConstructors.elm /Users/lydell/forks/elm-review/tests/MiscRules/NoUnusedTypeConstructorsTest.elm /Users/lydell/forks/elm-review/tests/MiscRules/ReadmeStartsWithProjectTitle.elm /Users/lydell/forks/elm-review/tests/MiscRules/ReadmeStartsWithProjectTitleTest.elm /Users/lydell/forks/elm-review/tests/ModuleNameLookupTableTest.elm /Users/lydell/forks/elm-review/tests/NoBooleanCaseOf.elm /Users/lydell/forks/elm-review/tests/NoBooleanCaseOfTest.elm /Users/lydell/forks/elm-review/tests/NoDebug/Log.elm /Users/lydell/forks/elm-review/tests/NoDebug/LogTest.elm /Users/lydell/forks/elm-review/tests/NoDebug/TodoOrToString.elm /Users/lydell/forks/elm-review/tests/NoDebug/TodoOrToStringTest.elm /Users/lydell/forks/elm-review/tests/NoExposingEverything.elm /Users/lydell/forks/elm-review/tests/NoExposingEverythingTest.elm /Users/lydell/forks/elm-review/tests/NoExtraBooleanComparison.elm /Users/lydell/forks/elm-review/tests/NoExtraBooleanComparisonTest.elm /Users/lydell/forks/elm-review/tests/NoFullyAppliedPrefixOperator.elm /Users/lydell/forks/elm-review/tests/NoFullyAppliedPrefixOperatorTest.elm /Users/lydell/forks/elm-review/tests/NoImportingEverything.elm /Users/lydell/forks/elm-review/tests/NoImportingEverythingTest.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/BadAlias.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/BadAliasSet.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/Config.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/Context.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/MissingAlias.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/MissingAliasSet.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/ModuleUse.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/Visitor.elm /Users/lydell/forks/elm-review/tests/NoInconsistentAliases/Visitor/Options.elm /Users/lydell/forks/elm-review/tests/NoListLiteralsConcat.elm /Users/lydell/forks/elm-review/tests/NoListLiteralsConcatTest.elm /Users/lydell/forks/elm-review/tests/NoMissingSubscriptionsCall.elm /Users/lydell/forks/elm-review/tests/NoMissingSubscriptionsCallTest.elm /Users/lydell/forks/elm-review/tests/NoMissingTypeAnnotation.elm /Users/lydell/forks/elm-review/tests/NoMissingTypeAnnotationInLetIn.elm /Users/lydell/forks/elm-review/tests/NoMissingTypeAnnotationInLetInTest.elm /Users/lydell/forks/elm-review/tests/NoMissingTypeAnnotationTest.elm /Users/lydell/forks/elm-review/tests/NoMissingTypeExpose.elm /Users/lydell/forks/elm-review/tests/NoMissingTypeExposeTest.elm /Users/lydell/forks/elm-review/tests/NoModuleOnExposedNames.elm /Users/lydell/forks/elm-review/tests/NoModuleOnExposedNames/Context.elm /Users/lydell/forks/elm-review/tests/NoNoOpMsg.elm /Users/lydell/forks/elm-review/tests/NoNoOpMsgTest.elm /Users/lydell/forks/elm-review/tests/NoRecursiveUpdate.elm /Users/lydell/forks/elm-review/tests/NoRecursiveUpdateTest.elm /Users/lydell/forks/elm-review/tests/NoRedundantConcat.elm /Users/lydell/forks/elm-review/tests/NoRedundantConcatTest.elm /Users/lydell/forks/elm-review/tests/NoTodoComment.elm /Users/lydell/forks/elm-review/tests/NoTodoCommentTest.elm /Users/lydell/forks/elm-review/tests/NoUnapprovedLicense.elm /Users/lydell/forks/elm-review/tests/NoUnapprovedLicenseTest.elm /Users/lydell/forks/elm-review/tests/NoUnused/CustomTypeConstructorArgs.elm /Users/lydell/forks/elm-review/tests/NoUnused/CustomTypeConstructorArgsTest.elm /Users/lydell/forks/elm-review/tests/NoUnused/CustomTypeConstructors.elm /Users/lydell/forks/elm-review/tests/NoUnused/CustomTypeConstructorsTest.elm /Users/lydell/forks/elm-review/tests/NoUnused/Dependencies.elm /Users/lydell/forks/elm-review/tests/NoUnused/DependenciesTest.elm /Users/lydell/forks/elm-review/tests/NoUnused/Exports.elm /Users/lydell/forks/elm-review/tests/NoUnused/ExportsTest.elm /Users/lydell/forks/elm-review/tests/NoUnused/Modules.elm /Users/lydell/forks/elm-review/tests/NoUnused/ModulesTest.elm /Users/lydell/forks/elm-review/tests/NoUnused/NonemptyList.elm /Users/lydell/forks/elm-review/tests/NoUnused/Parameters.elm /Users/lydell/forks/elm-review/tests/NoUnused/ParametersTests.elm /Users/lydell/forks/elm-review/tests/NoUnused/Patterns.elm /Users/lydell/forks/elm-review/tests/NoUnused/Patterns/NameVisitor.elm /Users/lydell/forks/elm-review/tests/NoUnused/PatternsTests.elm /Users/lydell/forks/elm-review/tests/NoUnused/Variables.elm /Users/lydell/forks/elm-review/tests/NoUnused/VariablesTest.elm /Users/lydell/forks/elm-review/tests/NoUselessSubscriptions.elm /Users/lydell/forks/elm-review/tests/NoUselessSubscriptionsTest.elm /Users/lydell/forks/elm-review/tests/Review/ExceptionsTest.elm /Users/lydell/forks/elm-review/tests/Review/Fix/FixTest.elm /Users/lydell/forks/elm-review/tests/Review/Rule/ElmJsonErrorsTest.elm /Users/lydell/forks/elm-review/tests/Review/Rule/ProjectVisitTest.elm /Users/lydell/forks/elm-review/tests/Review/Rule/VisitorsOrderTest.elm /Users/lydell/forks/elm-review/tests/Review/Rule/WithDeclarationListVisitorTest.elm /Users/lydell/forks/elm-review/tests/Review/Test/ExpectedErrorOrderTest.elm /Users/lydell/forks/elm-review/tests/Review/Test/FailureMessageTest.elm /Users/lydell/forks/elm-review/tests/Tests/NoInconsistentAliases.elm /Users/lydell/forks/elm-review/tests/Tests/NoModuleOnExposedNames.elm /Users/lydell/forks/elm-review/tests/Tests/NoModuleOnExposedNames/Context.elm /Users/lydell/forks/elm-review/tests/Tests/Vendor/NameVisitor.elm /Users/lydell/forks/elm-review/tests/Util.elm /Users/lydell/forks/elm-review/tests/Vendor/NameVisitor.elm --output /dev/null
Running /Users/lydell/forks/elm-review/node_modules/.bin/elm make /Users/lydell/forks/elm-review/elm-stuff/generated-code/elm-community/elm-test/0.19.1-revision4/src/Test/Generated/Main3603156898.elm --output /Users/lydell/forks/elm-review/elm-stuff/generated-code/elm-community/elm-test/0.19.1-revision4/elmTestOutput.js

elm-test 0.19.1-revision4
-------------------------

Running 618 tests. To reproduce these results, run: elm-test --fuzz 100 --seed 215189831069289


TEST RUN PASSED

Duration: 2151 ms
Passed:   618
Failed:   0
```

But since we added elm-json and got rid of elmi-to-json we only run the Elm compiler once:

```
❯ ~/forks/node-test-runner/bin/elm-test --verbose
Running /usr/local/bin/elm make /Users/lydell/forks/elm-review/elm-stuff/generated-code/elm-community/elm-test/0.19.1-revision4/src/Test/Generated/Main1263752250.elm --output /Users/lydell/forks/elm-review/elm-stuff/generated-code/elm-community/elm-test/0.19.1-revision4/elmTestOutput.js

elm-test 0.19.1-revision4
-------------------------

Running 618 tests. To reproduce these results, run: elm-test --fuzz 100 --seed 89275975664235


TEST RUN PASSED

Duration: 2276 ms
Passed:   618
Failed:   0
```

So all `--verbose` is doing now is saying that it’s compiling a generated file into an internal location. Not very useful.

Maybe a more useful `--verbose` flag could make a comeback in the future if we feel a need for it.

Reply:
Thanks for the explanation, that flag does look pretty useless. I have made a note to mention it in the release notes just to be on the safe side.